### PR TITLE
 librbd: sync flush should re-use existing async flush logic

### PR DESCRIPTION
**Pull request by dillaman**

### CODE REVIEW

File: src/librbd/internal.cc [hunk 1]
```diff
@@ -2221,28 +2221,6 @@ int validate_pool(IoCtx &io_ctx, CephContext *cct) {
...
-
-    ictx->user_flushed();
-    C_SaferCond ctx;
-    {
-      RWLock::RLocker owner_locker(ictx->owner_lock);
-      ictx->flush(&ctx);
-    }
-    r = ctx.wait();
-
-    ictx->perfcounter->inc(l_librbd_flush);
```
Comment (trociny) [inline src/librbd/internal.cc]: @dillaman Are you going to leave "flush" counter unused?

Comment (trociny) [reply src/librbd/internal.cc]: BTW, in "qa/workunits/rbd/test_admin_socket.sh" we have a test that checks "flush" perfcounter is updated after "flush" admin socket command.

Comment (dillaman) [reply src/librbd/internal.cc]: Yeah -- I thought it was already covered here [1] but it's for AIO flush. I'll combine them in the next push.

[1] https://github.com/ceph/ceph/blob/9b18093892b093daf09a3315e41da2a7f58ac351/src/librbd/io/ImageRequest.cc#L753

Comment (dillaman) [reply src/librbd/internal.cc]: ... consolidated

File: src/tools/rbd/action/Bench.cc [hunk 2]
```diff
@@ -207,6 +207,12 @@ int do_bench(librbd::Image& image, io_type_t io_type,
     return -EINVAL;
   }
 
+  int r = image.flush();
+  if (r < 0) {
+    std::cerr << "rbd: failed to flush: " << cpp_strerror(r) << std::endl;
+    return r;
```
Comment (trociny) [inline src/tools/rbd/action/Bench.cc]: Do we want this to be fatal? E.g. previously running bench read against exclusively mapped image would succeed.

Comment (dillaman) [reply src/tools/rbd/action/Bench.cc]: Tweaked it to be fatal for write/read-write tests and skipped on the final flush for read-only tests

**Comment by trociny** [comment]
@dillaman `qa/workunits/rbd/journal.sh` needs update. Namely, `test_rbd_journal()`, `rbd journal inspect` test, which counts journal entries after `rbd bench-write` and fails now because of 2 new "AioFlush" entries.

**Comment by dillaman** [comment]
@trociny updated

**Pull request merged**